idaes/core/control_volume0d.py

```diff
@@ -1630,12 +1630,11 @@ def calculate_scaling_factors(self):
         if hasattr(self, "rate_reaction_extent"):
             for (t, r), v in self.rate_reaction_extent.items():
                 if iscale.get_scaling_factor(v) is None:
-                    sf = iscale.get_scaling_factor(
-                        self.reactions[t].reaction_rate[r],
-                        default=1,
-                        warning=True)
-                    sf *= iscale.get_scaling_factor(self.volume[t])
-                    iscale.set_scaling_factor(v, sf)
+                    # Control Volume has no way of knowing how best to scale
+                    # reaction extent - e.g. whether it will be calculated from
+                    # reaction rate or a fixed yield. Unit models should
+                    # provide better scaling factors where possible.
+                    iscale.set_scaling_factor(v, 1)
```

Comment:
This is okay, but if you just want the scaling factor to be 1 you don't need to set it.  If you don't set it you'll be able to generate a warning based on it not being set.

Reply:
I actually did this somewhat deliberately as a way of guiding the user as to what variables existed but were not scaled. This way, the user can display the suffix and see that this variable has a factor of 1, whereas if it were not set it would not appear at all. Given that the control volumes are likely "black boxes" to many users, I thought it would help to populate the suffix as a guide.

Reply:
I know we need a better way to guide users to the variables that they should scale.  Either documentation or some sort of automated tools.  As you say, if we fill the suffixes of variables we expect need to be scaled, we could develop some sort of tool based on that, so I'm good.

Reply:
Some counter-points:

1. Regarding the warning system, the control volume will generally be one of the last things called - only the unit model and flowsheet should come later. If the unit model is depending on the control volume to calculate scaling factors for reaction extents, then the modeler has a fundamental disconnect in how things should be working that the warning won't help fix. That is - the unit model really should be telling the control volume what the scaling is and not the other way around. This is actually the key point here - the current scaling tools assume information only flows from the bottom up, where as there are cases like this where information has to flow both ways.
2. It is rather hard to document all of this, especially for something as generic as a control volume. Unit models definitely should recommend setting scaling factors for these when appropriate, and that would supersede anything the control volume does.
3. I would argue that displaying the scaling factors is critical - as it shows you exactly what is being scaled and how. Seeing a lot of ones in there should be a suggestion that maybe you should consider updating these.

Reply:
My thoughts on your points:
1. The current scaling implementation allows unit models to assign scaling factors at the control volume level (i.e. the unit model's calculate_scaling_factor method could have a self.control_volume.X assignment). You are right that the current implementation doesn't support the control volume calculating other scaling factors based on the unit model input, but I'd argue that for any control volume variables that need input from the unit model, the scaling should be set at the unit model method (and that the control volume should not set it at all, even if it is just to 1).
2. Yes the control volume is too general for the scaling documentation. Unit model documentation should state whether the user needs to provide a scaling factor for the unit model's control volume.
3. Yes displaying scaling factors is important, but I don't think displaying and looking at the ones should be the way to see whether a variable had its scaling set or not. We have tools to make a generator of the unscaled variables and constraints for users.

An additional thought:
Most of the time when we are setting scaling factors, we have an if statement that checks if it is not None, so that `calculate_scaling_factors` doesn't override user specified scaling factors. But if the control volume puts a 1 in for a scaling factor that needs to updated elsewhere, then that messes up our override feature. The line of code in the unit model will see a scaling factor has been assigned and skip it.